CHANGELOG.rst

```diff
@@ -7,16 +7,85 @@
 Changelog
 =========
 
+
 0.10.0 (unreleased)
 -------------------
 
 **Breaking change**
-- Removed the deprecated :func:`ndonnx.promote_nullable` function. Use :func:`ndonnx.additional.make_nullable` instead.
 
-**Array API compliance**
+- Removed the deprecated :func:`ndonnx.promote_nullable` function. Use :func:`ndonnx.extensions.make_nullable` instead.
+- The existing (private) tooling to define custom data types has been removed.
+- The :func:`ndonnx.Array.len` member function has been removed since it is neither defined by ``numpy.ndarray`` nor the Array-API.
+- :func:`ndonnx.Array.size` now returns `int | None` in accordance to with the Array-API rather than a :class:`~ndonnx.Array` instance.
+
+
+**Bug fixes**
+
+- The following functions now correctly take the ``keepdims`` argument into account:
+  - :meth:`ndonnx.Array.sum`
+  - :meth:`ndonnx.Array.prod`
+  - :meth:`ndonnx.Array.max`
+  - :meth:`ndonnx.Array.min`
+  - :meth:`ndonnx.Array.all`
+  - :meth:`ndonnx.Array.any`
 
-- ndonnx now supports the :func:`ndonnx.__array_namespace_info__` function from the Array API standard.
+
+**New features**
+
+- ndonnx gained partial support for `float16` data types.
+- The following functions are now exposed in the ndonnx namespace in accordance to the Array-API standard:
+  - :func:`ndonnx.__array_namespace_info__`
+  - :func:`ndonnx.maximum`
+  - :func:`ndonnx.minimum`
+  - :func:`ndonnx.meshgrid`
+  - :func:`ndonnx.moveaxis`
+  - :func:`ndonnx.tile`
+  - :func:`ndonnx.unstack`
+  - :func:`ndonnx.vecdot`
+- The newly added :mod:`ndonnx.extensions` module exposes the following functions:
+  - :func:`~ndonnx.extensions.datetime_to_year_month_day`
+  - :func:`~ndonnx.extensions.fill_null`
+  - :func:`~ndonnx.extensions.get_mask`
+  - :func:`~ndonnx.extensions.is_float_dtype`
+  - :func:`~ndonnx.extensions.is_integer_dtype`
+  - :func:`~ndonnx.extensions.is_nullable_dtype`
+  - :func:`~ndonnx.extensions.is_nullable_float_dtype`
+  - :func:`~ndonnx.extensions.is_nullable_integer_dtype`
+  - :func:`~ndonnx.extensions.is_numeric_dtype`
+  - :func:`~ndonnx.extensions.is_onnx_dtype`
+  - :func:`~ndonnx.extensions.is_signed_integer_dtype`
+  - :func:`~ndonnx.extensions.s_time_unit`
+  - :func:`~ndonnx.extensions.is_unsigned_integer_dtype`
+  - :func:`~ndonnx.extensions.isin`
+  - :func:`~ndonnx.extensions.make_nullable`
+  - :func:`~ndonnx.extensions.put`
+  - :func:`~ndonnx.extensions.shape` (deprecated in favor of :func:`ndonnx.Array.dynamic_shape`)
```

Comment:
It seems odd to move the function and immediately deprecate it. User code will break regardless, so might as well just delete the function entirely?

Reply:
I meant that there since we expect other intentional user-breaking changes in this release, it seemed like an opportunity to also immediately do so for `shape` rather than give a deprecation it in the new module.